inc/Config/QueryRunner/QueryRunner.php

```diff
@@ -205,6 +206,18 @@ public function execute( HttpQueryInterface $query, array $input_variables ): ar
 			}
 		}
 
+		// If the query is a search query and the search term is empty, return a
+		// collection with no results. This is to exit early and avoid making an
+		// unnecessary API call.
+		if ( ConfigRegistry::SEARCH_QUERY_KEY === $key && empty( $input_variables[ ConfigRegistry::SEARCH_QUERY_KEY ] ) ) {
+			return [
+				'is_collection' => true,
+				'metadata' => [],
+				'pagination' => [],
+				'results' => [],
+			];
+		}
+
```

Comment:
Instead, the query should mark the input variable as `'required' => true` and the UI should be updated to implement this logic. Query runner should stay ignorant of implementation details like this.
```suggestion
```

Reply:
I agree, that's why I commented this line to ensure it would start a discussion. This was a workaround for me to get this data source working, and by no means should be the way it's implemented for all the other data sources.

Realistically, the query runner should check to see if a required input variable is provided and error out if it's not provided. The UI should ensure that such a query is not run in the first place, and the APIs should 400 out. I'm looking into how to implement this within the UI.

Reply:
This has been fixed